lib/prompts.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+var _ = require("lodash");
+var async = require("async");
+var inquirer = require("inquirer");
+
+/**
+ * Prompt user for input, validate, and add derived fields to final data object.
+ *
+ * @param {Object}    init      Initialization configuration (`prompts`, `derived`)
+ * @param {Function}  callback  Calls back with `(err, data)`
+ * @returns {void}
+ */
+module.exports = function (init, callback) {
+  if (!init) { return callback(new Error("Invalid init object")); }
+
+  // Mutate `prompts` into appropriate format.
+  var prompts = init.prompts || [];
+  if (!_.isArray(prompts)) {
+    if (!_.isObject(prompts)) {
+      return callback(new Error("Invalid prompts type: " + typeof prompts));
+    }
+
+    prompts = _.map(prompts, function (val, key) {
+      return _.extend({ name: key }, val);
+    });
+  }
+
+  // Execute prompts, then derive final data.
+  async.auto({
+    prompts: function (cb) {
```

Comment:
The general rule is: `callback` is from the public method and preferred. Once I use that I can't have something _else_ named `callback` in my async* stuff, so I use `cb`